City-of-Helsinki/kukkuu-admin
# KK-677 | Add project select dropdown

## Description

Adds a project select dropdown which list the projects the user has access to. They are able to change between projects by using the select.

- If there's only a single project, the project should be shown as text
- If there're more than a single project, a dropdown should be rendered
- After selecting a project, the admin interface should refresh its data so that that the selected projects events etc. are shown
- By default the project with the most recent year should be selected

I also refactored the project utilities into a projectService for easier testing and maintenance.

## Context




[KK-677](https://helsinkisolutionoffice.atlassian.net/browse/KK-677)

## How Has This Been Tested?

I've added unit/integration level tests for the component.

## Manual Testing Instructions for Reviewers

As a user with access to a single project

1. Expect to see the year and name of the project on the AppBar on the top right corner of the page

As a user with access to many projects, with locale storage cleared

1. Expect the project with the recent most year to be selected by default
2. Expect to be able to open the menu by clicking
3. After selecting another project, expect the admin to refresh its data
